### PR TITLE
[wip] github actions to build the backend packages

### PR DESCRIPTION
**Pull request by FriedrichWilken**
This addresses [this issue](https://github.com/kubernetes-sigs/kpng/issues/50). It's an implementation of some simple GitHub actions that try to `go build` the backend packages. `go build` on packages discards any binaries immediately after, only giving feedback IF there was an issue while trying to build.

### CODE REVIEW

**Comment by FriedrichWilken** [comment]
TODO add win and macos
```    runs-on: ubuntu-latest ```

**Comment by mcluseau** [comment]
backends will mostly be OS-specific, so I think the matrix approach will not work for this test. But building `kpng`  should, as backends will simply be added in their os-specific `storecmd_xxx.go`.

**Comment by FriedrichWilken** [comment]
> backends will mostly be OS-specific, so I think the matrix approach will not work for this test. But building `kpng` should, as backends will simply be added in their os-specific `storecmd_xxx.go`.

I am not sure if I get this right. Do you say: don't try to build it on all OS'. Or is just this matrix approach not working?

**Comment by FriedrichWilken** [comment]
Because it is a little confusing to find the gh action results between different commits and branches: [this is the result of the test of the last commit](https://github.com/FriedrichWilken/kpng/actions/runs/1427932363)

File: .github/workflows/backend-build.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+    
+    - name: Build backends/nft
+      run: |
+        cd ./backends/nft
+        go mod vendor
+        go build
```
Comment (jayunit100) [inline .github/workflows/backend-build.yml:83]: imo this is a good start !

**Comment by jayunit100** [comment]
lets get these passing and then merge !

**Comment by jayunit100** [comment]
for now we can ignore windows, merge and iterate ! thank you friedrich !

**Comment by jayunit100** [comment]
as a FOLLOW ON can you or neha break the code out into a Makefile?

File: .github/workflows/backend-build.yml (hunk 1 above)
Comment (mcluseau) [reply .github/workflows/backend-build.yml:83]: wouldn't `go mod download` be enough? Sorry I tend to be a semantic nazi :p

**Comment by jayunit100** [comment]
Ok thanks for prototypeing now we know what to do #96

**Comment by FriedrichWilken** [comment]
This PR was made obsolete by @nehaLohia27 's [PR](https://github.com/kubernetes-sigs/kpng/pull/97) thus I will close it.

**Pull request closed without merging**